beamer/agent.py

```diff
@@ -31,107 +37,154 @@ def _get_deployment_block(contract_info: dict[str, ContractInfo]) -> BlockNumber
+    def _init_l1_chain(self) -> Chain:
+        l1_w3 = make_web3(self._config.rpc_urls["l1"], self._config.account)
+        l1_w3.middleware_onion.add(latest_block_based_cache_middleware)
+        chain_id = ChainId(l1_w3.eth.chain_id)
+        return Chain(w3=l1_w3, id=chain_id, rpc_url=self._config.rpc_urls["l1"])
+
+    def _init_rollups(self) -> dict[ChainId, Rollup]:
+        rollups: dict[ChainId, Rollup] = {}
+        for chain_name, rpc_url in self._config.rpc_urls.items():
+            if chain_name == "l1":
+                continue
+            w3 = make_web3(rpc_url, self._config.account)
+            chain_id = ChainId(w3.eth.chain_id)
+            if chain_id in rollups:
+                continue
+            contracts_info = _get_contracts_info(self._config, w3)
+            contracts = make_contracts(w3, contracts_info)
+            request_manager = contracts["RequestManager"]
+            fill_manager = contracts["FillManager"]
+            self._event_monitors[chain_id] = EventMonitor(
+                web3=w3,
+                contracts=(request_manager, fill_manager),
+                deployment_block=_get_deployment_block(contracts_info),
+                poll_period=POLL_PERIOD,
+                on_new_events=[],
+                on_sync_done=[],
+            )
+            rollups[chain_id] = Rollup(
+                w3=w3,
+                id=chain_id,
+                rpc_url=rpc_url,
+                contracts_info=contracts_info,
+                contracts=contracts,
+                request_manager=request_manager,
+                fill_manager=fill_manager,
+            )
+        return rollups
+
+    def _check_rollup(self, source_chain: Rollup, target_chain: Rollup) -> None:
+        max_validity_period = source_chain.request_manager.functions.MAX_VALIDITY_PERIOD().call()
 
         if self._config.unsafe_fill_time >= max_validity_period:
             raise RuntimeError(f"Unsafe fill time must be less than {max_validity_period}")
-        if not target_fill_manager.functions.allowedLps(self._config.account.address).call():
+        if not target_chain.fill_manager.functions.allowedLps(self._config.account.address).call():
             raise RuntimeError("Agent address is not whitelisted on FillManager")
-        if not source_request_manager.functions.allowedLps(self._config.account.address).call():
+        if not source_chain.request_manager.functions.allowedLps(
+            self._config.account.address
+        ).call():
             raise RuntimeError("Agent address is not whitelisted on RequestManager")
 
-        claim_request_extension = source_request_manager.functions.claimRequestExtension().call()
+    def _setup_direction(
+        self, direction: tuple[ChainId, ChainId], rollups: dict[ChainId, Rollup], l1: Chain
+    ) -> None:
+        source_chain = rollups[direction[0]]
+        target_chain = rollups[direction[1]]
+
+        self._check_rollup(source_chain, target_chain)
```

Comment:
I think we should split `_check_rollup` into two: `_check_source_rollup` and `_check_target_rollup`.
(Here we are providing both, which reads very weirdly.)
Then we could have
```python
        self._check_source_rollup(source_chain)
        self._check_target_rollup(source_chain)
```
Actually, even better would be 
```python
        self._check_source_chain(source_chain)
        self._check_target_chain(target_chain)
```
(this would require renaming `Rollup` to `Chain` and the base type `Chain` to something else (`ChainBase`?).

I feel the chain would also be more general since we're going to support Ethereum L1 in the short term and that is not really a rollup, so the "chain" naming would be more appropriate.